JSON Export

eedrummer opened the pull request:
Allows Synthea `Person` objects to be exported as JSON. Documentation on [JSON Export page of the Synthea wiki](https://github.com/synthetichealth/synthea/wiki/JSON-Export).

dehall commented on src/main/resources/synthea.properties:
```diff
@@ -27,7 +27,9 @@ exporter.practitioner.fhir_stu3.export = false
 exporter.encoding = UTF-8
 exporter.csv.export = false
-# if exporter.csv.append_mode = true, then each run will add new data to any existing CSVs. if false, each run will clear out the files and start fresh
+exporter.json.export = false
```
This is super pedantic but can these new properties be moved up or down so they don't split the CSV-specific properties?

dehall commented on src/main/java/org/mitre/synthea/export/JSONExporter.java:
```diff
@@ -0,0 +1,125 @@
+   * @return a lot of JSON in a String
+   */
+  public static String export(Person person) {
+    Gson gson = new GsonBuilder()
```
It might be nice to enable pretty printing on the resulting JSON, otherwise it all gets written to one line:
```suggestion
    Gson gson = new GsonBuilder()
            .setPrettyPrinting()
```

dehall commented on src/main/java/org/mitre/synthea/export/JSONSkip.java:
```diff
@@ -0,0 +1,12 @@
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.FIELD})
+public @interface JSONSkip {
+  // Marker interface indicating that a field should be skipped in the JSON export
```
This comment should be a Javadoc so that IDE users get a descriptive hint when they mouseover the interface

dehall commented on src/test/java/org/mitre/synthea/export/JSONExporterTest.java:
```diff
@@ -0,0 +1,34 @@
+public class JSONExporterTest {
+  @Rule
+  public TemporaryFolder tempFolder = new TemporaryFolder();
+
```
Could you also add a test for the "exporter.json.include_module_history" property? That seems like the kind of thing where someday a refactor could break that feature.

dehall commented on src/main/java/org/mitre/synthea/export/JSONExporter.java:
```diff
@@ -0,0 +1,125 @@
+        boolean keepEntry = true;
+        if (key.startsWith("ehr_") || key.contains("lookup") || key.contains("UUID")) {
+          keepEntry = false;
+        } else if ((!Config.getAsBoolean("exporter.json.include_module_history"))
```
Seems like a good flag to cache as synthea patients have a lot of attributes

dehall reviewed (commented):
Looks good, just a few quick suggestions

eedrummer commented:
@dehall I think I have addressed all of you comments.

Also:
* Fixed an issue with Java 17 compatibility (may impact previous versions as well)
* Moved the JSON export testing to the parallel testing tools that were added with the test suite performance improvements

dehall commented:
Looks good, will merge pending the TestHelper change we discussed separately.
